planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -221,25 +227,74 @@ ObjectDataArray AvoidanceModule::calcAvoidanceTargetObjects(
 
-    // Object is on center line -> ignore.
-    if (std::abs(object_data.lateral) < parameters_.threshold_distance_object_is_on_center) {
-      DEBUG_PRINT("Ignore object: (object is on center line)");
+    // Find the footprint point closest to the path, set to object_data.overhang_distance.
+    object_data.overhang_dist =
+      calcOverhangDistance(object_data, object_closest_pose, object_data.overhang_pose.position);
+
+    lanelet::ConstLanelet overhang_lanelet;
+    if (!rh->getClosestLaneletWithinRoute(object_closest_pose, &overhang_lanelet)) {
       continue;
     }
 
-    // Find the footprint point closest to the path, set to object_data.overhang_distance.
-    object_data.overhang_dist = calcOverhangDistance(object_data, object_closest_pose);
+    if (overhang_lanelet.id()) {
+      object_data.overhang_lanelet = overhang_lanelet;
+      lanelet::BasicPoint3d overhang_basic_pose(
+        object_data.overhang_pose.position.x, object_data.overhang_pose.position.y,
+        object_data.overhang_pose.position.z);
+      if (isOnRight(object_data)) {
+        const auto & target_left_line = [this, &rh, &overhang_lanelet]() {
+          if (
+            parameters_.enable_avoidance_over_same_direction &&
+            parameters_.enable_avoidance_over_opposite_direction) {
+            return rh->getLeftMostLinestring(overhang_lanelet);
+          } else if (
+            parameters_.enable_avoidance_over_same_direction &&
+            !parameters_.enable_avoidance_over_opposite_direction) {
+            return rh->getLeftMostSameDirectionLinestring(overhang_lanelet);
+          }
+          return overhang_lanelet.leftBound();
+        }();
+        object_data.to_road_shoulder_distance =
+          distance2d(to2D(overhang_basic_pose), to2D(target_left_line.basicLineString()));
+        debug_linestring.push_back(target_left_line);
+      } else {
+        const auto & target_right_line = [this, &rh, &overhang_lanelet]() {
+          if (
+            parameters_.enable_avoidance_over_same_direction &&
+            parameters_.enable_avoidance_over_opposite_direction) {
+            return rh->getRightMostLinestring(overhang_lanelet);
+          } else if (
+            parameters_.enable_avoidance_over_same_direction &&
+            !parameters_.enable_avoidance_over_opposite_direction) {
+            return rh->getRightMostSameDirectionLinestring(overhang_lanelet);
+          }
+          return overhang_lanelet.rightBound();
+        }();
+        object_data.to_road_shoulder_distance =
+          distance2d(to2D(overhang_basic_pose), to2D(target_right_line.basicLineString()));
+        debug_linestring.push_back(target_right_line);
+      }
+    }
 
     DEBUG_PRINT(
-      "set object_data: longitudinal = %f, lateral = %f, largest_overhang = %f",
-      object_data.longitudinal, object_data.lateral, object_data.overhang_dist);
+      "set object_data: longitudinal = %f, lateral = %f, largest_overhang = %f, "
```

Comment:
```suggestion
      "set object_data: longitudinal = %f, lateral = %f, largest_overhang = %f", 
```

Reply:
@zulfaqar-azmi-t4 
if you press commit suggestion you can apply it alternetively